prometheus-extension-ballerina/metric_reporter.bal

```diff
@@ -59,45 +60,39 @@ isolated function startReporter(string host, int port) returns error? {
-                            foreach var aSnapshot in summaries {
-                                tags[EXPIRY_TAG] = aSnapshot.timeWindow.toString();
-                                payload += generateMetricHelp(qualifiedMetricName, "A Summary of " +  qualifiedMetricName + " for window of "
-                                                            + aSnapshot.timeWindow.toString());
-                                payload += generateMetricInfo(qualifiedMetricName, METRIC_TYPE_SUMMARY);
-                                payload += generateMetric(getMetricName(qualifiedMetricName, "mean"), tags, aSnapshot.mean);
-                                payload += generateMetric(getMetricName(qualifiedMetricName, "max"), tags, aSnapshot.max);
-                                payload += generateMetric(getMetricName(qualifiedMetricName, "min"), tags, aSnapshot.min);
-                                payload += generateMetric(getMetricName(qualifiedMetricName, "stdDev"), tags,
-                                aSnapshot.stdDev);
-                                foreach var percentileValue in aSnapshot.percentileValues  {
-                                    tags[PERCENTILE_TAG] = percentileValue.percentile.toString();
-                                    payload += generateMetric(qualifiedMetricName, tags, percentileValue.value);
-                                }
-                                _ = tags.remove(EXPIRY_TAG);
-                                _ = tags.remove(PERCENTILE_TAG);
+                        observe:Snapshot[] summaries = <observe:Snapshot[]> metric.summary;
+                        foreach var aSnapshot in summaries {
+                            tags[EXPIRY_TAG] = aSnapshot.timeWindow.toString();
+                            payload.push(generateMetricHelp(qualifiedMetricName, string `A Summary of ${
+                                                        qualifiedMetricName} for window of ${aSnapshot.timeWindow.toString()}`));
+                            payload.push(generateMetricInfo(qualifiedMetricName, METRIC_TYPE_SUMMARY));
+                            payload.push(generateMetric(getMetricName(qualifiedMetricName, "mean"), tags, aSnapshot.mean));
+                            payload.push(generateMetric(getMetricName(qualifiedMetricName, "max"), tags, aSnapshot.max));
+                            payload.push(generateMetric(getMetricName(qualifiedMetricName, "min"), tags, aSnapshot.min));
+                            payload.push(generateMetric(getMetricName(qualifiedMetricName, "stdDev"), tags, aSnapshot.stdDev));
+                            foreach var percentileValue in aSnapshot.percentileValues  {
+                                tags[PERCENTILE_TAG] = percentileValue.percentile.toString();
+                                payload.push(generateMetric(qualifiedMetricName, tags, percentileValue.value));
                             }
+                            _ = tags.remove(EXPIRY_TAG);
```

Comment:
Do we need this remove op, as there will be a new value replacing the existing value before any read (L:77)

Reply:
Drop removing the `EXPIRY_TAG` with 6ec19022df8d407b37ebc8fa0de0ef588ce4400a.
However, we need to keep `_ = tags.remove(PERCENTILE_TAG);`, otherwise, it will add `quantile` incorrectly.